packages/yarnpkg-core/sources/scriptUtils.ts

```diff
@@ -142,22 +151,24 @@ export async function makeScriptEnv({project, locator, binFolder, lifecycleScrip
 const MAX_PREPARE_CONCURRENCY = 2;
 const prepareLimit = pLimit(MAX_PREPARE_CONCURRENCY);
 
-export async function prepareExternalProject(cwd: PortablePath, outputPath: PortablePath, {configuration, report, workspace = null}: {configuration: Configuration, report: Report, workspace?: string | null}) {
+export async function prepareExternalProject(cwd: PortablePath, outputPath: PortablePath, {configuration, report, workspace = null, locator}: {configuration: Configuration, report: Report, workspace?: string | null, locator: Locator}) {
   await prepareLimit(async () => {
     await xfs.mktempPromise(async logDir => {
       const logFile = ppath.join(logDir, `pack.log` as Filename);
 
       const stdin = null;
       const {stdout, stderr} = configuration.getSubprocessStreams(logFile, {prefix: cwd, report});
 
-      const packageManager = await detectPackageManager(cwd);
+      stdout.write(`Installing the external project "${locator.name}" from sources\n\n`);
+
+      const packageManagerSelection = await detectPackageManager(cwd);
       let effectivePackageManager: PackageManager;
 
-      if (packageManager !== null) {
-        stdout.write(`Installing the project using ${packageManager}\n\n`);
-        effectivePackageManager = packageManager;
+      if (packageManagerSelection !== null) {
+        stdout.write(`Installing the external project "${locator.name}" using ${packageManagerSelection.packageManager}. Reason: ${packageManagerSelection.reason}\n\n`);
+        effectivePackageManager = packageManagerSelection.packageManager;
       } else {
-        stdout.write(`No package manager detected; defaulting to Yarn\n\n`);
+        stdout.write(`No package manager detected for external project "${locator.name}"; defaulting to Yarn\n\n`);
```

Comment:
```suggestion
        stdout.write(`No package manager detected for external project "${structUtils.prettyLocator(locator)}"; defaulting to Yarn\n\n`);
```

Reply:
Replaced your call by `stringifyLocator`, since it'll be written into a file

Reply:
Updated